sentry_sdk/integrations/beam.py

```diff
@@ -113,63 +111,56 @@ def _wrap_task_call(func):
     # type: (F) -> F
     """
     Wrap task call with a try catch to get exceptions.
-    Pass the client on to raise_exception so it can get rebinded.
     """
-    client = Hub.current.client
 
     @wraps(func)
     def _inner(*args, **kwargs):
         # type: (*Any, **Any) -> Any
         try:
             gen = func(*args, **kwargs)
         except Exception:
-            raise_exception(client)
+            raise_exception()
 
         if not isinstance(gen, types.GeneratorType):
             return gen
-        return _wrap_generator_call(gen, client)
+        return _wrap_generator_call(gen)
 
     setattr(_inner, USED_FUNC, True)
     return _inner  # type: ignore
 
 
-def _capture_exception(exc_info, hub):
-    # type: (ExcInfo, Hub) -> None
+def _capture_exception(exc_info):
+    # type: (ExcInfo) -> None
     """
     Send Beam exception to Sentry.
     """
-    integration = hub.get_integration(BeamIntegration)
-    if integration is None:
-        return
+    client = sentry_sdk.get_client()
 
-    client = hub.client
-    if client is None:
+    integration = client.get_integration(BeamIntegration)
+    if integration is None:
```

Comment:
It would be nice if we could use `ensure_integration_enabled` here, but I don't think it is possible with the current implementation. I think I could tweak the decorator a bit to make this use case possible though

Reply:
Yes, this is something that would be useful in many other integrations as well. I think we could just make the function in the decorator optional.

Reply:
It is a bit more complex than that if we want type hinting to work properly. I am almost ready with a PR

Reply:
Here is the PR: #2893

Reply:
I just merged #2893. If you could update this branch and use the updated decorator here, that would be great!